arcgis-ios-sdk-samples/Layers/Edit KML ground overlay/EditKMLGroundOverlayViewController.swift

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class EditKMLGroundOverlayViewController: UIViewController {
+    // MARK: Storyboard views and properties
+    @IBOutlet var sceneView: AGSSceneView! {
+        didSet {
+            sceneView.scene = makeScene()
+        }
+    }
+    // The slider that controls the overlay's opacity.
+    @IBOutlet var opacitySlider: UISlider!
+    // The label that displays the slider's value.
+    @IBOutlet var valueLabel: UILabel!
+    
+    // The KML ground overlay.
+    var overlay: AGSKMLGroundOverlay?
+    
+    // MARK: Actions and methods
+    @IBAction func sliderValueChanged(_ slider: UISlider) {
+        // Change the color of the overlay according to the slider's value.
+        let alpha = CGFloat(slider.value)
+        overlay?.color = UIColor(red: 0, green: 0, blue: 0, alpha: alpha)
+        // Update the slider's value label.
+        valueLabel.text = String(format: "%.2f", slider.value)
+    }
+    
+    func makeScene() -> AGSScene {
+        // Create a scene for the scene view.
+        let scene = AGSScene(basemap: .imagery())
+        // Create a geometry for the ground overlay.
+        let overlayGeometry = AGSEnvelope(xMin: -123.066227926904, yMin: 44.04736963555683, xMax: -123.0796942287304, yMax: 44.03878298600624, spatialReference: .wgs84())
+        // Create a KML icon for the overlay image.
+        let imageURL = URL(string: "https://libapps.s3.amazonaws.com/accounts/55937/images/1944.jpg")!
+        let overlayImage = AGSKMLIcon(url: imageURL)
+        // Apply a KML ground overlay.
+        overlay = AGSKMLGroundOverlay(geometry: overlayGeometry, icon: overlayImage)
```

Comment:
Per its doc
> A new initialized ground overlay or nil if an unsupported geometry is passed.

It seems this initializer will always succeed given that the geometry is fixed. Maybe consider force unwrap here?

Reply:
If I force unwrap there and allow `overlay` to possibly be nil, won't it cause problems afterward? I'll also have to unwrap `overlay` about 5 times.

Reply:
I'm suggesting sth like
```swift
// Implicitly unwrap here.
var overlay: AGSKMLGroundOverlay!

// Since we know the initializer won't fail, it is safe to unwrap ❗  it.
let overlay = AGSKMLGroundOverlay(geometry: overlayGeometry, icon: overlayImage)!
// Assign the newly created object to the instance property.
self.overlay = overlay
```

But it totally up to your choice. It is fine to keep as-is. 😁  I prefer to bang it whenever it is safe.